OverlayPlugin.Core/Integration/FFXIVRepository.cs

```diff
@@ -274,6 +275,19 @@ public string GetLocaleString()
             }
         }
 
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        internal bool WriteLogLineImpl(uint ID, string line)
+        {
+            var plugin = GetPluginData();
+            if (plugin == null) return false;
+            var iocContainer = plugin.pluginObj.GetType().GetField("_iocContainer", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(plugin.pluginObj);
+            var getServiceMethod = iocContainer.GetType().GetMethod("GetService");
+            var logOutput = getServiceMethod.Invoke(iocContainer, new object[] { typeof(FFXIV_ACT_Plugin.Logfile.ILogOutput) }) as FFXIV_ACT_Plugin.Logfile.ILogOutput;
```

Comment:
Ravahn suggested in discord that you should use reflection/dynamic here: "adding a direct reference to FFXIV_ACT_Plugin.Logfile.dll will eventually be a problem.  Either use pure reflection, or use dynamic keyword and wrap it with try/catch exception so that mismatches dont stop everything from working."

I still also think it'd be better (if we can) to get this once and cache it rather than doing this reflection dance on every write.

Reply:
I suspect you can do this sort of thing `dynamic logOutput = getServiceMethod.Invoke(iocContainer, new object[] { getServiceMethod.ReturnType });` to avoid the DLL reference on that line, but my C# isn't strong enough to know off the top of my head about how to convert a uint to an enum known only at runtime via reflection.

Reply:
Yeah, I poked around at this for a couple hours and couldn't get `dynamic` to work, so I just switched to flat-out using reflection for the whole thing.